### PR TITLE
Add typings to the inline completer example

### PR DESCRIPTION
**Pull request by jtpio**
## References

Add typings to the inline completer example: https://jupyterlab.readthedocs.io/en/latest/extension/extension_points.html#inline-completer

To make easier to copy and paste the example, and have the extension build directly.

## Code changes

Documentation update.

## User-facing changes

None

## Backwards-incompatible changes

None

### CODE REVIEW

**Review by krassowski — DISMISSED** [review]
Thank you @jtpio!

File: docs/source/extension/extension_points.rst [hunk 1]
```diff
@@ -905,16 +905,24 @@ interface allows for much more extensive customization of the completer.
 
 .. code:: typescript
 
-    import { ICompletionProviderManager, ICompletionProvider } from '@jupyterlab/completer';
+    import {
+      CompletionHandler,
+      ICompletionProviderManager,
+      IInlineCompletionContext,
+      IInlineCompletionProvider
```
Comment (krassowski) [inline docs/source/extension/extension_points.rst:912]:
```suggestion
      ICompletionProviderManager,
      ICompletionContext,
      ICompletionProvider
```

File: docs/source/extension/extension_points.rst [hunk 2]
```diff
@@ -905,16 +905,24 @@ interface allows for much more extensive customization of the completer.
...
+    } from '@jupyterlab/completer';
 
     class MyProvider implements ICompletionProvider {
       readonly identifier = 'my-provider';
 
-      async isApplicable(context) {
+      async isApplicable(context: IInlineCompletionContext) {
```
Comment (krassowski) [inline docs/source/extension/extension_points.rst:918]:
```suggestion
      async isApplicable(context: ICompletionContext) {
```

File: docs/source/extension/extension_points.rst [hunk 3]
```diff
@@ -905,16 +905,24 @@ interface allows for much more extensive customization of the completer.
...
         return true;
       }
 
-      async fetch(request, context) {
+      async fetch(
+        request: CompletionHandler.IRequest,
+        context: IInlineCompletionContext
```
Comment (krassowski) [inline docs/source/extension/extension_points.rst:924]:
```suggestion
        request: CompletionHandler.IRequest,
        context: ICompletionContext
```

**Review by krassowski — CHANGES_REQUESTED** [review]
Sorry, I was too quick at first pass. I think you wanted to make the changes in `Inline Completer` section but you made them in `Code Completer` section. I added suggestions on the modified lines of code but GitHub UI does not allow me to add suggestions lower down on the `Inline Completer` section

**Comment by jtpio** [comment]
Ah yes good catch, it was one section below!

Fixing that now.

**Comment by jtpio** [comment]
Thanks for the review!

**Comment by jtpio** [comment]
@meeseeksdev please backport to 4.2.x

**Pull request merged**